### DISCUSSION TITLE
How do I implement update for a table row and some other rows connected to it via one-to-many connections?

### DISCUSSION THREAD

**Discussion by joulev** [prisma/prisma#16806] [Q&A] [answered]
I have this schema

```prisma
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider     = "mysql"
  url          = env("DATABASE_URL")
  relationMode = "prisma"
}

enum Language {
  EN
  FR
}

model Tag {
  id         String    @id @default(cuid())
  slug       String    @unique @db.VarChar(31)
  translated TagTr[]
}

model TagTr {
  id          String   @id @default(cuid())
  slug        String   @db.VarChar(31)
  tag         Tag      @relation(fields: [slug], references: [id])
  lang        Language
  name        String   @db.VarChar(31)
  description String   @db.VarChar(255)

  @@unique([slug, lang])
}
```

and use this code to update a `Tag`, or create a new `Tag` if `slug` doesn't exist:

```ts
// slug:        string,
// name:        { en: string, fr: string },
// description: { en: string, fr: string },

await prisma.tag.upsert({
  where: { slug },
  update: {
    translated: {
      update: [
        {
          where: { slug_lang: { slug, lang: "EN" } },
          data: { name: name.en || undefined, description: description.en || undefined },
        },
        {
          where: { slug_lang: { slug, lang: "FR" } },
          data: { name: name.en || undefined, description: description.fr || undefined },
        },
      ],
    },
  },
  create: {
    slug,
    translated: {
      create: [
        { lang: "EN", name: name.en, description: description.en },
        { lang: "FR", name: name.fr, description: description.fr },
      ],
    },
  },
});
```

But this fails with this error, even when I supply to this Prisma call a `slug` that's already existed in the database with both the `EN` `TagTr` and `FR` `TagTr` being available and connected.

```
An operation failed because it depends on one or more records that were required but not found.
No 'TagTr' record was found for a nested update on relation 'TagToTagTr'.
```

It is very confusing to me, as the `TagTr` records *should* be already available? I feel like I missed something *very* obvious here, what should I do to implement the behaviour?

Also I'm sorry if these code examples are too minimal to reproduce the issue, as I'm not sure how to create a minimal example in this case; also I believe my mistake is probably clear enough that more experienced Prisma users can look at the code and tell what's wrong.

<details>
<summary>Prisma v4.7.1, see <code>prisma --version</code> output</summary>

```
prisma                  : 4.7.1
@prisma/client          : 4.7.1
Current platform        : darwin
Query Engine (Node-API) : libquery-engine 272861e07ab64f234d3ffc4094e32bd61775599c (at ../../../../Library/pnpm/store/v3/tmp/dlx-17730/node_modules/.pnpm/@prisma+engines@4.7.1/node_modules/@prisma/engines/libquery_engine-darwin.dylib.node)
Migration Engine        : migration-engine-cli 272861e07ab64f234d3ffc4094e32bd61775599c (at ../../../../Library/pnpm/store/v3/tmp/dlx-17730/node_modules/.pnpm/@prisma+engines@4.7.1/node_modules/@prisma/engines/migration-engine-darwin)
Introspection Engine    : introspection-core 272861e07ab64f234d3ffc4094e32bd61775599c (at ../../../../Library/pnpm/store/v3/tmp/dlx-17730/node_modules/.pnpm/@prisma+engines@4.7.1/node_modules/@prisma/engines/introspection-engine-darwin)
Format Binary           : prisma-fmt 272861e07ab64f234d3ffc4094e32bd61775599c (at ../../../../Library/pnpm/store/v3/tmp/dlx-17730/node_modules/.pnpm/@prisma+engines@4.7.1/node_modules/@prisma/engines/prisma-fmt-darwin)
Format Wasm             : @prisma/prisma-fmt-wasm 4.7.1-1.272861e07ab64f234d3ffc4094e32bd61775599c
Default Engines Hash    : 272861e07ab64f234d3ffc4094e32bd61775599c
Studio                  : 0.477.0
```

</details>

**Comment by joulev** [comment, marked as answer]
Sorry, it was indeed the case I made a super dumb mistake. My relation field should have been `@relation(fields: [slug], references: [slug])` instead.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
